### ISSUE TITLE
How can I exec host command when deploying the clab?

### ISSUE THREAD

**Issue by lingdudd** [srl-labs/containerlab#1478]
I want to exec command to configure the route to the container in host when starting the lab. Is there any way to define the commands in the yml file?

**Comment by hellt** [comment]
Not in the yml file. You will have to create a shell script that adds that route before or after calling the 'containerlab deploy'

**Comment by steiler** [comment]
Yes there is. Simply add a node of kind "host" and use the exec parameter.
If it is about the network configuration you can also spin up a container of kind linux and and set the namespace for this container to the host namespace and adjust the networking via the container. (https://containerlab.dev/manual/nodes/#network-mode)
BTW @hellt we seem to be missing a host kind documentation entry.

**Comment by hellt** [comment]
@steiler did you try that?
Iirc exec is not implemented for host kind

**Comment by steiler** [comment]
Yes I had used this before. If it is no longer working then we broke something.
It might not be implemented specifically via the host kind, but the host namespace identifier "__host" or so it was, is used in the Network namespace library. AFAIR

**Comment by steiler** [comment]
WARN[0000] Exec operation is not implemented for kind "host"

**Comment by steiler** [comment]
``> WARN[0000] Exec operation is not implemented for kind "host" Let me double check if we can delegate to default_node and remove the warning.

Ok, we must have changed stuff, running this via node "host" won't work. I'll reimplement that.
But anyways, this will work
```
  l2:
      kind: linux
      image: alpine:latest
      network-mode: host
      exec:
        - ip r
        - ip l
```

**Comment by steiler** [comment]
Exec on host kind implemented in PR #1481
